### PR TITLE
Cadence:HiFi Mini NN Library: Optimized kernels integration

### PR DESCRIPTION
**Pull request by nyadla-sys**
Integrated fully_connected, softmax and svdf cadence  HiFi Mini optimized kernels
to tensor flow lite micro kernels. These optimized kernels gives better performance
over reference kernels while running keyword_benchmark application.

Signed-off-by: Bhanu Prakash Bandaru Venkata <bhanup@cadence.com>
Signed-off-by: Harshavardhan Ravindra Joshi <joshih@cadence.com>
Signed-off-by: Pramod Kumar Surana <pramods@cadence.com>
Signed-off-by: Prasad Nikam <pnikam@cadence.com>

### CODE REVIEW

**Comment by gbaned** [comment]
@nyadla-sys Thank you for your contribution. Can you please sign CLA? Thanks!

**Comment by gbaned** [comment]
@nyadla-sys Can you please sign CLA and resolve conflicts? Thanks!

**Review by njeffrie — APPROVED** [review]
We have internally moved xtensa_hifimini production kernels to xtensa_hifimini_legacy to make room for these to land.  Once everything is worked out W.R.T CLA and presubmit (you'll have to sync to HEAD), we are ready to accept these and begin internal testing.

Moving forward, we will move these to xtensa_hifimini_staging and move our production kernels back to xtensa_hifimini.  That way we can merge these kernels one-by-one after validating, and align with internal deadlines.  I am creating a RFC document to outline the proposal for landing staged changes: https://docs.google.com/document/d/1tUvZUXPq-gW1F6uYWOUGkUuJ_DXseWsH3u_QdvTKI-g/edit?usp=sharing

**Comment by rthadur** [comment]
@nyadla-sys can you please check latest comments from @njeffrie

**Comment by nyadla-sys** [comment]
@rthadur @njeffrie 
Bhanu uploaded new PR and I am going to take off this pull request
https://github.com/tensorflow/tensorflow/pull/39691
